utils/build-script-impl

```diff
@@ -922,138 +1121,156 @@ if [[ ! "${SKIP_BUILD_SWIFTPM}" ]] ; then
+                test_this_target=$(not ${SKIP_TEST_IOS_SIMULATOR})
+                ;;
+            appletvos-*)
+                swift_sdk="TVOS"
+                build_for_this_target=$(not ${SKIP_BUILD_TVOS_DEVICE})
+                if [[ ! "${SKIP_TEST_TVOS_HOST}" ]] ; then
+                    test_host_only=1
+                else
+                    test_this_target=
+                fi
+                build_benchmark_this_target=$(not ${SKIP_BUILD_TVOS_DEVICE})
+                ;;
+            appletvsimulator-*)
+                swift_sdk="TVOS_SIMULATOR"
+                build_for_this_target=$(not ${SKIP_BUILD_TVOS_SIMULATOR})
+                test_this_target=$(not ${SKIP_TEST_TVOS_SIMULATOR})
+                ;;
+            watchos-*)
+                swift_sdk="WATCHOS"
+                build_for_this_target=$(not ${SKIP_BUILD_WATCHOS_DEVICE})
+                if [[ ! "${SKIP_TEST_WATCHOS_HOST}" ]] ; then
+                    test_host_only=1
+                else
+                    test_this_target=
+                fi
+                build_benchmark_this_target=$(not ${SKIP_BUILD_WATCHOS_DEVICE})
+                ;;
+            watchsimulator-*)
+                swift_sdk="WATCHOS_SIMULATOR"
+                build_for_this_target=$(not ${SKIP_BUILD_WATCHOS_SIMULATOR})
+                test_this_target=$(not ${SKIP_TEST_WATCHOS_SIMULATOR})
+                ;;
+            android-*)
+                swift_sdk="ANDROID"
+                build_for_this_target=$(not ${SKIP_BUILD_ANDROID})
+                # FIXME: Allow Android host tests to be enabled/disabled by the
+                #        build script.
                 test_this_target=
-            fi
-            build_benchmark_this_target=$(not ${SKIP_BUILD_WATCHOS_DEVICE})
-            ;;
-        watchsimulator-*)
-            build_for_this_target=$(not ${SKIP_BUILD_WATCHOS_SIMULATOR})
-            test_this_target=$(not ${SKIP_TEST_WATCHOS_SIMULATOR})
-            ;;
-        android-*)
-            build_for_this_target=$(not ${SKIP_BUILD_ANDROID})
-            # FIXME: Allow Android host tests to be enabled/disabled by the
-            #        build script.
-            test_this_target=
-            ;;
-        *)
-            echo "Unknown compiler deployment target: ${deployment_target}"
-            exit 1
-            ;;
-    esac
-    if [[ "${build_for_this_target}" ]] ; then
-        if [[ "${BUILD_SWIFT_STDLIB_UNITTEST_EXTRA}" == "1" ]] ; then
-            SWIFT_STDLIB_TARGETS+=("swift-stdlib-${deployment_target}")
-        else
-            if [[ "${VALIDATION_TEST}" == "1" || "${LONG_TEST}" == "1" ]] ; then
-                SWIFT_STDLIB_TARGETS+=("swift-stdlib-${deployment_target}")
+                ;;
+            *)
+                echo "Unknown compiler deployment target: ${stdlib_deployment_target}"
+                exit 1
+                ;;
+        esac
+
+        SWIFT_SDKS+=("${swift_sdk}")
```

Comment:
> OK I've tested that we can build on OSX and it configures for all available SDKs but only builds OSX.

I tested your branch with:

```
$ ./utils/build-script -R --distcc --build-stdlib-deployment-targets=macosx-x86_64
```

And I see build-script-impl configuring Swift like this:

```
-DSWIFT_SDKS:STRING=IOS;IOS_SIMULATOR;OSX;TVOS;TVOS_SIMULATOR;WATCHOS;WATCHOS_SIMULATOR
```

I think you need to wrap everything starting from this line until the end of the loop in `if [ "${is_in_build_list}" ]]`.  This would get us the desired behavior.